src/cryptonote_core/loki_name_system.cpp

```diff
@@ -894,14 +961,14 @@ CREATE TABLE IF NOT EXISTS "settings" (
 CREATE TABLE IF NOT EXISTS "mappings" (
     "id" INTEGER PRIMARY KEY NOT NULL,
     "type" INTEGER NOT NULL,
-    "name" VARCHAR NOT NULL,
-    "value" BLOB NOT NULL,
+    "name_hash" BLOB NOT NULL,
+    "encrypted_value" BLOB NOT NULL,
     "txid" BLOB NOT NULL,
     "prev_txid" BLOB NOT NULL,
     "register_height" INTEGER NOT NULL,
     "owner_id" INTEGER NOT NULL REFERENCES "owner" ("id")
 );
-CREATE UNIQUE INDEX IF NOT EXISTS "name_type_id" ON mappings("name", "type");
+CREATE UNIQUE INDEX IF NOT EXISTS "name_hash_type_id" ON mappings("name_hash", "type");
```

Comment:
I think the change to BLOB here is a potential performance issue because of the unique index: BLOBs aren't indexable in the same way a VARCHAR is; doing any sort of comparison (including on insertion to check the unique constraint) will most likely end up needing to do a full table scan to check against every existing blob value.

I'd suggest that we keep `name_hash` as a VARCHAR and store it as hex (or perhaps base64-encoded?) for performance reasons.  It'll use a bit more space, but will keep `name_hash` lookups collatable (and thus binary-searchable).

Reply:
Ok, have changed. I defer to your expertise on this sorta stuff.